eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/immutable/AbstractImmutableBiMap.java

```diff
@@ -483,4 +484,67 @@ protected Object writeReplace()
             return new ImmutableBiMapSerializationProxy<>(this);
         }
     }
+
+    protected static class ImmutableKeySet<K> extends AbstractImmutableSet<K>
```

Comment:
@johnjstuart don't forget to override contains too because it is implemented in `AbstractRichIterable` in the hierarchy!

Reply:
@johnjstuart no because it goes through the whole collection in O(n) while in this case, you can do it in O(1) since you are in a map implementation.

Reply:
Added a `contains` implementation.